### PR TITLE
CSP-17 Get standards from courses api

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.Roatp.CourseManagement/InnerApi/Requests/GetStandardRequest.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.Roatp.CourseManagement.InnerApi.Requests
+{
+    public class GetStandardRequest : IGetApiRequest
+    {
+        public string GetUrl => $"/api/courses/Standards/{StandardCode}";
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.CourseManagement/InnerApi/Requests/GetStandardRequest.cs:7]: good spot, I checked against other apim repos (eg Commitments etc) and it doesn't have it, will take off the prefix

File: src/SFA.DAS.Roatp.CourseManagement.Api/Startup.cs [hunk 2]
```diff
@@ -45,6 +46,7 @@ public void ConfigureServices(IServiceCollection services)
 
                 services.AddAuthentication(azureAdConfiguration, policies);
             }
+            services.AddMediatR(GetType().Assembly, typeof(GetProviderCourseQueryHandler).Assembly);
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Api/Startup.cs:49]: There is no need to do this.

File: src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourseQuery/GetProviderCourseQueryHandler.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            _courseManagementApiClient = courseManagementApiClient;
+            _logger = logger;
+        }
+
+        public async Task<GetProviderCourseResult> Handle(GetProviderCourseQuery request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Get Provider Course request received for ukprn {ukprn}, LarsCode {larsCode}", request.Ukprn, request.LarsCode);
+
+            GetStandardResponse standard;
+            try
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourseQuery/GetProviderCourseQueryHandler.cs:31]: Remove the higher level try catch, this should be handled by a middleware

File: src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourseQuery/GetProviderCourseQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                    return null;
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Error occurred trying to retrieve standard for LarsCode {request.LarsCode}");
+                throw;
+            }
+
+            try
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourseQuery/GetProviderCourseQueryHandler.cs:46]: Remove higher level try catch

File: src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourseQuery/GetProviderCourseQueryHandler.cs (hunk 3 above)
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourseQuery/GetProviderCourseQueryHandler.cs:31]: I thought we'd established the middleware in apim doesn't do this, so we have to leave this in.  I'll dive in and have a look

**Pull request merged**